src/app/login/page.tsx

```diff
@@ -0,0 +1,81 @@
+"use client";
+
+import { signIn } from "next-auth/react";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import image from "./logo.png";
+import Image from 'next/image'
+import styles from "./styles.module.css";
+import { PasswordField } from "@/components/ButtonPassword/PasswordField";
+
+const LoginPage = () => {
+  const [errors, setErrors] = useState<string[]>([]);
+  const [email, setEmail] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
```

Comment:
another error here.
ok, all this code can be changed using another approach let me explain to you.
When you are coding try to think of the best way to write the code following the DRY principles, in this case, you need to take the data from the forms. in this case, you can create a new hook for example called useForm.ts
```
import { useState, ChangeEvent } from 'react';

type FormState = { [key: string]: any };

const useForm = (initForm: FormState) => {
    const [form, setForm] = useState<FormState>(initForm);

    const handlerChangeForm = ({ target }: ChangeEvent<HTMLInputElement>) => {
        setForm({...form, [target.name]: target.value});
    }

    const handlerResetForm = () => {
        setForm(initForm);
    }

    return [form, handlerChangeForm, handlerResetForm];
}

export default useForm;
```
And you can use it.

```
import React, { FC } from 'react';
import useForm from './useForm'; // Asume que useForm está en el mismo directorio

const UserForm: FC = () => {
    const initialFormState = { name: '', email: '' };
    const [form, handlerChangeForm, handlerResetForm] = useForm(initialFormState);

    const handleSubmit = (event: React.FormEvent) => {
        event.preventDefault();
        console.log('Form Submitted:', form);
        handlerResetForm();
    }

    return (
        <form onSubmit={handleSubmit}>
            <div>
                <label htmlFor="name">Name:</label>
                <input 
                    type="text" 
                    id="name" 
                    name="name" 
                    value={form.name} 
                    onChange={handlerChangeForm} 
                />
            </div>
            <div>
                <label htmlFor="email">Email:</label>
                <input 
                    type="email" 
                    id="email" 
                    name="email" 
                    value={form.email} 
                    onChange={handlerChangeForm} 
                />
            </div>
            <button type="submit">Submit</button>
        </form>
    );
}

export default UserForm;
```
please refactor de code following this approach